### PR TITLE
Fetch from client

### PR DESCRIPTION
**Pull request by diversifiedmechanic**
I kind of got sidetracked because I realized that I need the fetch to work before I really could start diving into the React Components.  Tests are included.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -7,14 +7,21 @@ app.use(express.static('./public'));
...
+  } else {
+    // need to sanitize this at some point
+    db.getAllSellerItemsExceptCurrentItem(req.params.id)
+      .then((allItems) => {
+        res.status(200).json(allItems);
+      })
+      .catch(() => {
+        res.status(404).json({ error: 'That ID does not exist in the database' });
+      });
+  }
```
Comment (naszzz) [inline server/index.js]: that's cool the way your handling the conditional check to see if req param is a number. but yea, this looks good to me.

File: server/index.test.js [hunk 2]
```diff
@@ -36,9 +36,16 @@ describe('API endpoint with /:id', () => {
...
+  it('should return with a 400 error if sent an invalid input', (done) => {
     endpoint
       .get('/Foobar')
+      .expect(400)
+      .end(done);
+  });
+
+  it('should return with a 404 error if the ID is not in the database', (done) => {
+    endpoint
+      .get('/100000')
```
Comment (naszzz) [inline server/index.test.js]: so this is pretty cool too. didn't know testing enpoints could be done this way. i had been doing something along these lines ....  request(app).get('/1').then((response) => {
      expect(response.statusCode).toBe(200);
      done(); ... 

so the .then(cb) is not actually needed.. that's cool.

File: src/lib.test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  it('should handle recieving nothing in the body', async () => {
+    fetchMock.mock('/10000', {
+      body: JSON.stringify({ error: 'That ID does not exist in the database' }),
+      status: 200,
+    });
+    const blankFoo = await getAdditionalItems(10000);
+    expect(blankFoo.error).toEqual('That ID does not exist in the database');
+  });
+});
```
Comment (naszzz) [inline src/lib.test.js]: all this is pretty badass.

File: server/index.js (hunk 1 above)
Comment (diversifiedmechanic) [reply server/index.js]: I think it's pretty hacky.  It works but there are better ways of doing it, I'm just not sure what it is.

File: server/index.test.js (hunk 2 above)
Comment (diversifiedmechanic) [reply server/index.test.js]: This is going back to getting hanging tests.  I think both are valid but you have to choose one or the other.  If you use promises, you can't use the `end`.

File: src/lib.test.js (hunk 3 above)
Comment (diversifiedmechanic) [reply src/lib.test.js]: Thanks!

**Pull request merged**